tests/connection/test_fsm.py

```diff
@@ -0,0 +1,139 @@
+    rsm.reserve_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveChecking.value
+    rsm.reserve_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveHeld.value
+    rsm.reserve_commit_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveCommitting.value
+    rsm.reserve_commit_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    #
+    # reserve_request -> reserve_confirmed -> reserve_commit_request -> reserve_commit_failed
+    #
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    rsm.reserve_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveChecking.value
+    rsm.reserve_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveHeld.value
+    rsm.reserve_commit_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveCommitting.value
+    rsm.reserve_commit_failed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    #
+    # reserve_request -> reserve_confirmed -> reserve_timeout_notification -> reserve_commit_request -> reserve_commit_confirmed
+    #
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    rsm.reserve_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveChecking.value
+    rsm.reserve_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveHeld.value
+    rsm.reserve_timeout_notification()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveTimeout.value
+    rsm.reserve_commit_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveCommitting.value
+    rsm.reserve_commit_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    #
+    # reserve_request -> reserve_confirmed -> reserve_timeout_notification -> reserve_abort_request -> reserve_abort_confirmed
+    #
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    rsm.reserve_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveChecking.value
+    rsm.reserve_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveHeld.value
+    rsm.reserve_timeout_notification()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveTimeout.value
+    rsm.reserve_abort_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveAborting.value
+    rsm.reserve_abort_confirmed()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+
+
+def test_provision_state_machine() -> None:  # noqa: D103
```

Comment:
Perhaps add a github action?

Reply:
Good idea. I will follow your suggestion and add this in another PR.